### PR TITLE
Add supported custom font formats in official docs

### PR DESCRIPTION
**Pull request by bharat-biradar**
This PR adds the custom font formats that are supported by flutter in the text style docs.

Issues - #76715 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by HansMuller** [comment]
CC @GaryQian @jason-simmons

File: packages/flutter/lib/src/painting/text_style.dart [hunk 1]
```diff
@@ -334,6 +334,14 @@ const double _kDefaultFontSize = 14.0;
 /// ```
 /// {@end-tool}
 ///
+/// #### Supported font formats
+///
+/// Font formats currently supported by Flutter :
+/// * `.ttf`
```
Comment (GaryQian) [inline packages/flutter/lib/src/painting/text_style.dart:340]: Flutter also supports TrueType Collections, (`.ttc`) which contain multiple tts files. This is commonly seen in CJK font collections that include multiple fonts for multiple languages.

Comment (GaryQian) [reply packages/flutter/lib/src/painting/text_style.dart:340]: For example, see https://www.google.com/get/noto/help/cjk/ where the truetype downloads are `.ttc` files.

**Comment by GaryQian** [comment]
@jason-simmons are there any other more obscure formats that we support that I forgot?

**Comment by jason-simmons** [comment]
I think those are all the formats that Flutter promises to support.  Anything beyond that depends on the configuration of third-party libraries like FreeType.

This change selected the types of fonts included in the engine's build of FreeType:
https://flutter.googlesource.com/third_party/freetype2/+/a10b062df0c8958d69377aa04ea6554a9961a111

File: packages/flutter/lib/src/painting/text_style.dart (hunk 1 above)
Comment (bharat-biradar) [reply packages/flutter/lib/src/painting/text_style.dart:340]: @GaryQian I had missed to mention it. Thanks for the tip. I'll make the changes.

File: packages/flutter/lib/src/painting/text_style.dart [hunk 2]
```diff
@@ -334,6 +334,15 @@ const double _kDefaultFontSize = 14.0;
...
+/// * `.otf`
+/// * `.ttc`
```
Comment (GaryQian) [inline packages/flutter/lib/src/painting/text_style.dart:342]: very minor nit: Move ttc to be next to ttf for better organization

**Review by GaryQian — APPROVED** [review]
Thanks!

File: packages/flutter/lib/src/painting/text_style.dart [hunk 3]
```diff
@@ -334,6 +334,15 @@ const double _kDefaultFontSize = 14.0;
 /// ```
 /// {@end-tool}
 ///
+/// #### Supported font formats
+///
+/// Font formats currently supported by Flutter :
+/// * `.ttf`
```
Comment (GaryQian) [inline packages/flutter/lib/src/painting/text_style.dart:340]: add additional space before each `*`

File: packages/flutter/lib/src/painting/text_style.dart [hunk 4]
```diff
@@ -334,6 +334,15 @@ const double _kDefaultFontSize = 14.0;
 /// ```
 /// {@end-tool}
 ///
+/// #### Supported font formats
+///
+/// Font formats currently supported by Flutter :
```
Comment (GaryQian) [inline packages/flutter/lib/src/painting/text_style.dart:339]: Add newline after this line.

Comment (bharat-biradar) [reply packages/flutter/lib/src/painting/text_style.dart:339]: @GaryQian I have committed the proposed changes. Please check if there are any further changes to be made.

Thank you.

Comment (GaryQian) [reply packages/flutter/lib/src/painting/text_style.dart:339]: It LGTM now, thank you for your contribution!

Comment (bharat-biradar) [reply packages/flutter/lib/src/painting/text_style.dart:339]: @GaryQian the cookbook doesn't mention .ttc too. Should it also be updated?

**Pull request merged**